### ISSUE TITLE
cpu206C2_plat03_ver0000001F_2018-05-08_PRD_77DADA73 could not be inserted with MMTOOL

### ISSUE THREAD

**Issue by ms178** [platomav/CPUMicrocodes#7]
Hey platomav! I've downloaded the mentioned microcode from your database and tried to insert it into my customized BIOS with MMTool, but the tool throws an error claiming that the *.bin is not a valid microcode file. Could you please fix it? I remember a similar issue with the 1D microcode which another user fixed with a Hexeditor (back then in early 2017). Could you please fix it and re-upload? I am afraid but I am not able to fix it myself.

**Comment by ms178** [comment]
Additional info: The former two insertable microcode files were 10 kB in size. This one has 11 kB currently.

**Comment by platomav** [comment]
The microcodes are all proper. Read why below.

https://github.com/platomav/MCExtractor/issues/18
https://github.com/platomav/MCExtractor/issues/3

**Comment by LSSoniX** [comment]
Microcode is correct, but not even. Try to align the microcode to even. Add a 1024-byte PAD file filled with FF,

**Comment by platomav** [comment]
It is as LSSoniX said but it also depends on the BIOS. Some may require 0x400, 0x800, 0x1000 etc alignment as usually defined in a GUID at the end of the BIOS. It was explained by DeathBringer at some overclockers post a few months ago. Either way, as I said, that is an AMI/MMTool specific message and quirk so it doesn't concern the general microcode repository.

**Comment by ms178** [comment]
@platomav Sorry for the noise and my very limited understanding for the inner working of these things. It just happens to hold me back to mod my Asus P6TWS and P6X58D-E BIOS. Is there a guide somewhere which details every step? To have these fixed files with the correct alignement would be of great help for people like me, but I understand that this could be out of the scope of this database.

@LSSoniX 

Sorry, but could you dumb it down for me a little more? I did open the file with an online Hexeditor. How many lines of FF should I add? I could try to analyze both the newest file and the older known working file and see where this leads me. Thanks for your help, I appreciate it!

**Comment by LSSoniX** [comment]
@ms178 
Can you give a link to the original BIOS file?

**Comment by ms178** [comment]
@LSSoniX  Thank you very much, but I just figured it out myself. It turned out that the Asus BIOS wanted a fixed size of 13800 Bytes, so I 00'ed it all up to this size (it took me some time to get to that conclusion). Et voila, it worked. Just flashed it and confirmed it in Windows seconds ago.

**Comment by LSSoniX** [comment]
@ms178 
If 00 is filled, then this is not correct, it is necessary FF,

**Comment by ms178** [comment]
@LSSoniX 
I took the 00's as I had analyzed a previous working microcode file which did exactly that. And as it works for me now (confirmed through a look into the registry where Windows shows me that it loads the 1F microcode file from BIOS), I don't see the need to tinker with it further. What would be the difference in using FFs?
